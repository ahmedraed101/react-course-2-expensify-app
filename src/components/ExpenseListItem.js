import React from 'react'
import moment from 'moment'
import { Link } from 'react-router-dom'
import { formatToDollar } from './../utils/formaters'

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <div className="container">
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
            <p>
                {formatToDollar(amount)}
                <br />
                {moment(createdAt).format('LL')}
                {/* MMM Do, YYYY = LL */}
            </p>
        </div>
    )
}

export default ExpenseListItem
