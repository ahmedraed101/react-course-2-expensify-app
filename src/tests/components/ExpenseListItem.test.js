import React from 'react'
import { shallow } from 'enzyme'
import expenses from './../fixtures/expenses'
import ExpenseListItem from './../../components/ExpenseListItem'

test('should render expense list item with values', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot()
})
