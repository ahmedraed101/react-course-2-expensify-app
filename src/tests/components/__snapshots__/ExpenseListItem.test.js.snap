// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item with values 1`] = `
<div
  className="container"
>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $30.00
    <br />
    January 1, 1970
  </p>
</div>
`;
